beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -1564,49 +1560,51 @@ impl<T: BeaconChainTypes> ExecutionPendingBlock<T> {
-                        }
-                    }
-                    if !consensus_context.blobs_verified_vs_txs()
-                        && verify_kzg_commitments_against_transactions::<T::EthSpec>(
-                            transactions,
-                            kzg_commitments,
-                        )
-                        //FIXME(sean) we should maybe just map this error so we have more info about the mismatch
-                        .is_err()
+        if let Some(data_availability_boundary) = chain.data_availability_boundary() {
+            if block_slot.epoch(T::EthSpec::slots_per_epoch()) >= data_availability_boundary {
+                let sidecar = block
+                    .blobs()
+                    .ok_or(BlockError::BlobValidation(BlobError::MissingBlobs))?;
+                let kzg = chain.kzg.as_ref().ok_or(BlockError::BlobValidation(
+                    BlobError::TrustedSetupNotInitialized,
+                ))?;
+                let transactions = block
+                    .message()
+                    .body()
+                    .execution_payload_eip4844()
+                    .map(|payload| payload.transactions())
+                    .map_err(|_| BlockError::BlobValidation(BlobError::TransactionsMissing))?
+                    .ok_or(BlockError::BlobValidation(BlobError::TransactionsMissing))?;
+                let kzg_commitments = block
+                    .message()
+                    .body()
+                    .blob_kzg_commitments()
+                    .map_err(|_| BlockError::BlobValidation(BlobError::KzgCommitmentMissing))?;
+                if !consensus_context.blobs_sidecar_validated() {
+                    if !kzg_utils::validate_blobs_sidecar(
+                        &kzg,
+                        block.slot(),
+                        block_root,
+                        kzg_commitments,
+                        sidecar,
+                    )
+                    .map_err(|e| BlockError::BlobValidation(BlobError::KzgError(e)))?
                     {
-                        return Err(BlockError::BlobValidation(
-                            BlobError::TransactionCommitmentMismatch,
-                        ));
+                        return Err(BlockError::BlobValidation(BlobError::InvalidKzgProof));
                     }
                 }
+                if !consensus_context.blobs_verified_vs_txs()
+                    && verify_kzg_commitments_against_transactions::<T::EthSpec>(
+                        transactions,
+                        kzg_commitments,
+                    )
+                    //FIXME(sean) we should maybe just map this error so we have more info about the mismatch
+                    .is_err()
+                {
+                    return Err(BlockError::BlobValidation(
+                        BlobError::TransactionCommitmentMismatch,
+                    ));
+                }
```

Comment:
sorry this is  a big diff but the only thing changed is rather than using the consensus context to check for a glob, we try to get it off the block wrapper and send an error if we don't have it when we think we should 

So this out block was removed
`if let Some(ref sidecar) = consensus_context.blobs_sidecar() {}`

And this was added
```
               let sidecar = block
                    .blobs()
                    .ok_or(BlockError::BlobValidation(BlobError::MissingBlobs))?;
```

Reply:
ty <3